File: isegoria/src/main/java/com/eulersbridge/isegoria/profile/badges/BadgeViewHolder.java
package com.eulersbridge.isegoria.profile.badges;

import android.graphics.Color;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.RequestManager;
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
import com.eulersbridge.isegoria.R;
import com.eulersbridge.isegoria.network.api.models.Badge;

class BadgeViewHolder extends RecyclerView.ViewHolder {

    private Badge item;

    final private ImageView imageView;
    final private TextView nameTextView;
    final private TextView descriptionTextView;

    private @Nullable RequestManager glide;

    BadgeViewHolder(View view) {
        super(view);

        imageView = view.findViewById(R.id.badge_list_image_view);
        nameTextView = view.findViewById(R.id.badge_list_name_text_view);
        descriptionTextView = view.findViewById(R.id.badge_list_description_text_view);
    }

    void setItem(@Nullable Badge item, boolean completed) {
        this.item = item;

        if (item == null) {
            nameTextView.setText(null);
            descriptionTextView.setText(null);
            imageView.setImageDrawable(null);

        } else {
            if (!completed) {
                imageView.setColorFilter(Color.argb(125, 35, 35, 35));
            } else {
                imageView.clearColorFilter();
            }

            nameTextView.setText(item.name);
            descriptionTextView.setText(item.description);
            imageView.setImageDrawable(null);
        }
    }

    void setImageUrl(@NonNull RequestManager glide, long itemId, @NonNull String imageUrl) {
        if (item != null && item.id == itemId) {

            this.glide = glide;

            glide.load(imageUrl)
                    .transition(DrawableTransitionOptions.withCrossFade())
                    .into(imageView);
        }
    }

    void onRecycled() {
        if (glide != null)
            glide.clear(imageView);
    }
}

File: isegoria/src/main/kotlin/com/eulersbridge/isegoria/profile/badges/BadgeViewHolder.kt
package com.eulersbridge.isegoria.profile.badges

import android.graphics.Color
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.RequestManager
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.eulersbridge.isegoria.R
import com.eulersbridge.isegoria.network.api.model.Badge

internal class BadgeViewHolder(view: View) : RecyclerView.ViewHolder(view) {

    private var item: Badge? = null
    private var glide: RequestManager? = null

    private val imageView: ImageView = view.findViewById(R.id.badge_list_image_view)
    private val nameTextView: TextView = view.findViewById(R.id.badge_list_name_text_view)
    private val descriptionTextView: TextView = view.findViewById(R.id.badge_list_description_text_view)

    fun setItem(item: Badge?, completed: Boolean) {
        this.item = item

        bindItem(item, completed)
    }

    private fun bindItem(item: Badge?, completed: Boolean) {
        if (item == null) {
            nameTextView.text = null
            descriptionTextView.text = null
            imageView.setImageDrawable(null)

        } else {
            if (completed) {
                imageView.clearColorFilter()
            } else {
                imageView.setColorFilter(Color.argb(125, 35, 35, 35))
            }

            nameTextView.text = item.name
            descriptionTextView.text = item.description
            imageView.setImageDrawable(null)
        }
    }

    fun setImageUrl(glide: RequestManager, itemId: Long, imageUrl: String) {
        if (item?.id == itemId) {
            this.glide = glide

            imageView.post {
                glide.load(imageUrl)
                        .transition(DrawableTransitionOptions.withCrossFade())
                        .into(imageView)
            }
        }
    }

    fun onRecycled() {
        glide?.clear(imageView)
    }
}
